### PR TITLE
MSUnmerged: support detail and rse_type in the RESTful APIs

### LINKED ISSUE #10692: MSUnmerged: Implement data/status and data/info interfaces to the service
**Impact of the new feature**
MSUnmerged

**Is your feature request related to a problem? Please describe.**
In the process of implementing the MSUnmeged service we did follow the old ways of creating reports as it was done in all other micro services. Meaning one single report per service. In this case, unfortunately, this approach is not going to work since we are having three different instances of the service but they all get their own names and deployment cycles. They share only a common MSConfig. So the links bellow simply won't work:

https://cmsweb.cern.ch/ms-unmerged/data/info 
https://cmsweb.cern.ch/ms-unmerged/data/status

And this was also the reason why we had to disable the `livenessprobe` check during the Kubernetes deployment in production - https://github.com/dmwm/WMCore/issues/10682#issuecomment-884897414 .

We need  to develop a proper way to expose those interfaces for the 3 different instances of the service with the proper counters and reports. The links to them should be:

https://cmsweb.cern.ch/ms-unmerged-t1/data/info 
https://cmsweb.cern.ch/ms-unmerged-t1/data/status

https://cmsweb.cern.ch/ms-unmerged-t2-t3/data/info 
https://cmsweb.cern.ch/ms-unmerged-t2-t3/data/status

https://cmsweb.cern.ch/ms-unmerged-t2-t3-us/data/info 
https://cmsweb.cern.ch/ms-unmerged-t2-t3-us/data/status

Or alternatively having one report f all three in a single page. Which in my opinion would be the best but since those are independent services that would be quite difficult to synchronize.

It would  be great if we can also report some counters on a per RSE base.

**Describe the solution you'd like**
Develop the proper interfaces to every instance of the service with the full set of reports.  And also reflect all of them in the full K8 setup of the service. 

**Describe alternatives you've considered**
None

**Additional context**
None

### PR DESCRIPTION
**Pull request by amaltaro**
Fixes #10692 

#### Status
ready

#### Description
This PR provides the following changes:
* fix `stop` call to the MSUnmerged thread
* affecting **all** microservices: support keyword args for the `info` API. For MSUnmerged, add `rse_type` key/value pair to the response, in case it was provided by the user.
* affecting **all** microservices: support keyword args for the `status` API. For MSUnmerged, add `rse_type` key/value pair to the response, in case it was provided by the user. If `detail=true` is provided, then also return a summary of the last MSUnmerged cycle results.

Second commit contains a fix for MSUnmerged and MSRuleCleaner error message reported through the `status` REST API.

#### Is it backward compatible (if not, which system it affects?)
yes

#### Related PRs
None

#### External dependencies / deployment changes
None

### CODE REVIEW

**Comment by amaltaro** [comment]
These two REST APIs have been tested for all the microservices. A couple of issues have been spotted and fixed as well.

File: test/python/WMCore_t/MicroService_t/MicroService_t.py [hunk 1]
```diff
@@ -24,14 +24,20 @@ def __init__(self, config=None):
...
+        return data
 
-    def info(self, reqName):
+    def info(self, reqName, **kwargs):
         "Return current status about our service"
         print("### CALL info API with request name %s" % reqName)
+        data = {'status': "OK", "api": "info"}
+        if kwargs:
+            data.update(kwargs)
         return {'status': "OK", "api": "info"}
```
Comment (todor-ivanov) [inline test/python/WMCore_t/MicroService_t/MicroService_t.py:41]: Shouldn't you `return data` from here too?

**Comment by amaltaro** [comment]
Merging it now to request a new testbed deployment. Comments/questions and review is still welcome though.

**Review by todor-ivanov — CHANGES_REQUESTED** [review]
Thanks  @amaltaro  I left few minor things in line. All are minor and can be skipped, but one (the last one) of them may need your attention.

File: src/python/WMCore/MicroService/MSManager.py [hunk 2]
```diff
@@ -288,16 +288,27 @@ def stop(self):
...
         return status
 
-    def info(self, reqName=None):
+    def info(self, reqName=None, **kwargs):
         """
         Return transfer information for a given request
         :param reqName: request name
+        :param kwargs: other arguments that can be provided for different
+            microservices. Examples:
+            rse_type: string with the type of the RSEs to fetch (MSUnmerged)
```
[the remark this answers is not in the document]
Comment (amaltaro) [reply src/python/WMCore/MicroService/MSManager.py:303]: Good idea. Adding.

File: src/python/WMCore/MicroService/MSManager.py [hunk 3]
```diff
@@ -288,16 +288,27 @@ def stop(self):
...
         :return: data transfer information for this request
         """
-        data = {"request": reqName, "transferDoc": None}
+        if 'ruleCleaner' in self.services or 'unmerged' in self.services:
+            data = {}
```
[the remark this answers is not in the document]
Comment (amaltaro) [reply src/python/WMCore/MicroService/MSManager.py:307]: I wouldn't call it a bug, but request and transferDoc are meaningless for MSRuleCleaner, at the moment.

File: src/python/WMCore/MicroService/MSManager.py [hunk 4]
```diff
@@ -308,18 +319,25 @@ def info(self, reqName=None):
...
-    def status(self, detail):
+    def status(self, detail, **kwargs):
         """
         Return the current status of a MicroService and a summary
         of its last execution activity.
         :param detail: boolean used to retrieve some extra information
           regarding the service
+        :param kwargs: other arguments that can be provided for different
+            microservices. Examples:
+            rse_type: string with the type of the RSEs to fetch (MSUnmerged)
```
[the remark this answers is not in the document]
Comment (amaltaro) [reply src/python/WMCore/MicroService/MSManager.py:340]: Done

File: test/python/WMCore_t/MicroService_t/MicroService_t.py (hunk 1 above)
Comment (amaltaro) [reply test/python/WMCore_t/MicroService_t/MicroService_t.py:41]: haa, good catch!

File: src/python/WMCore/MicroService/MSManager.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (todor-ivanov) [reply src/python/WMCore/MicroService/MSManager.py:308]: agreed thanks

**Review by todor-ivanov — APPROVED** [review]
Things look good to me now .

Thanks @amaltaro

**Comment by amaltaro** [comment]
Thanks for the prompt review, Todor.

**Pull request merged**